src/wallet/connection.ts

```diff
@@ -88,4 +91,95 @@ class WalletServiceConnection extends BaseConnection {
   }
 }
 
-export default WalletServiceConnection;
+/**
+ * This connection is used when the wallet has websocket disabled.
+ */
+export class DummyWalletServiceConnection extends EventEmitter implements IConnection {
+  public readonly isDummyConnection = true;
+  public walletId?: string;
+
+  private currentNetwork: string;
+  private currentServer: string;
+  protected state: ConnectionState;
+
+  constructor(options: WalletServiceConnectionParams = { walletId: 'dummy-wallet' }) {
+    super();
+
+    const {
+      walletId,
+      network,
+      servers,
+    } = {
+      ...DEFAULT_PARAMS,
+      ...options,
+    };
+
+    if (!network) {
+      throw Error('You must explicitly provide the network.');
+    }
+
+    this.walletId = walletId;
+    this.currentNetwork = network || 'dummynet';
+    this.state = ConnectionState.CLOSED;
+    this.currentServer = servers[0] || config.getServerUrl();
+  }
+
+  start(): void {
+    if (!this.walletId) {
+      throw new Error('Wallet id should be set before connection start.');
+    }
+  }
+
+  stop(): void {
+    this.removeAllListeners();
+    this.setState(ConnectionState.CLOSED);
+  }
+
+  endConnection(): void {
+    // There is no connection to end
+  }
+
+  setup(): void {
+    // There is nothing to setup
+  }
+
+  handleWalletMessage(_wsData: any): void {
+    // There is nothing to handle
+  }
+
+  onConnectionChange(value: boolean): void {
+    if (value) {
+      this.setState(ConnectionState.CONNECTED);
+    } else {
+      this.setState(ConnectionState.CONNECTING);
+    }
+  }
```

Comment:
Do you really need to re-implement this method? Where is this used?

Reply:
As this is a simple state change, it is cheap to implement, and make the dummy usage more consistent overall.